src/dummy_http_server.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""This dummy HTTP server has been implemented to trick Prometheus Configurer. It returns
+`200 OK` status on any POST call.
+Prometheus Configurer was originally designed to share a volume with the Prometheus Server. After
+pushing new alert rules to the shared volume, Prometheus Configurer reloads the configuration
+of the Prometheus Server.
+Since prometheus-k8s and prometheus-configurer-k8s charms do not share a volume and also
+Prometheus Server delivered by the prometheus-k8s charm reloads automatically upon new rules
+detection, reload done by the Prometheus Configurer needs to be suppressed or otherwise it
+interferes with Prometheus Server's automatic reload, causing relaod failure.
+This dummy server's address is given to the `prometheus_configurer` service as the `prometheusURL`
+parameter, so that Prometheus Configurer thinks it reloaded Prometheus Server's configuration.
+"""
+
+import logging
+import os
+import subprocess
+import sys
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+from ops.charm import CharmBase
+from ops.framework import Object
+
+logger = logging.getLogger(__name__)
+
+LOG_FILE_PATH = "/var/log/prometheus-configurer-dummy-server.log"
+
+
+class DummyHTTPServer(Object):
+    def __init__(self, charm: CharmBase, port: int):
+        super().__init__(charm, None)
+        self._port = port
+
+    def start_server(self):
+        """Starts dummy HTTP server in a new process."""
+        logger.info(f"Starting dummy HTTP server on port {self._port}.")
```

Comment:
For the http server, shouldn't we have another container for it and start it via pebble

Reply:
I don't know. It's super lightweight and for most of time it'll stay idle (how often do you change the rules?). Does it make sense to spin up another container for something that is in fact a workaround?

Reply:
Well it's not exactly a workaround as it will stay this way. I'd rather have us configure services only via 1 way (i.e. pebble). We can use this [project](https://github.com/canonical/200-ok/pull/1), add a new container and have its pebble layer simply be 
```
Layer(
            {
                "summary": f"{self.service_name} pebble layer",
                "services": {
                    self.service_name: {
                        "override": "replace",
                        "startup": "enabled",
                        "command": "nginx",
                    }
                },
            }
        )
```

Reply:
Done.